chain/client/src/view_client.rs

```diff
@@ -1203,14 +1203,16 @@ impl Handler<NetworkViewClientMessages> for ViewClientActor {
 }
 
 impl Handler<GetGasPrice> for ViewClientActor {
-    type Result = Result<GasPriceView, String>;
+    type Result = Result<GasPriceView, GetGasPriceError>;
 
     #[perf]
     fn handle(&mut self, msg: GetGasPrice, _ctx: &mut Self::Context) -> Self::Result {
         let header = self
             .maybe_block_id_to_block_hash(msg.block_id)
             .and_then(|block_hash| self.chain.get_block_header(&block_hash));
-        header.map(|b| GasPriceView { gas_price: b.gas_price() }).map_err(|e| e.to_string())
+        header
+            .map(|b| GasPriceView { gas_price: b.gas_price() })
+            .map_err(|e| GetGasPriceError::from(e))
```

Comment:
Given we now have the implementation for `From<near_chain_primitives::Error> for GetGasPriceError`, we can use `?` operator, and simplify the code to the following:

```suggestion
        Ok(GasPriceView { gas_price: header?.gas_price() })
```